File: Electronics_Shop_17.Services/Database/Seller.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Services.Database
{
    public class Seller : IHelperId, ISoftDelete
    {
        public int Id { get; set; }
        public string StoreName { get; set; }
        public int LicenseNumber { get; set; }
        public int AdressId { get; set; }
        public virtual Adress Adress { get; set; }
        public int UserAccountId { get; set; }
        public virtual UserAccount UserAccount { get; set; }
        public int PersonId { get; set; }
        public virtual Person Person { get; set; }
        public bool isDeleted { get; set; }

    }
}

File: angular17_web/src/app/models/seller/seller.ts
import { Adress } from '../adress/adress';
import { Person } from '../person/person';

export class Seller {
  id!: number;
  storeName!: string;
  licenseNumber!: number;
  adress!: Adress;
  userAccountId!: number;
  person!: Person;
  isDeleted!: boolean;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.storeName = data.storeName;
    this.licenseNumber = data.licenseNumber;
    this.adress = data.adress ? new Adress(data.adress) : undefined as any;
    this.userAccountId = data.userAccountId;
    this.person = data.person ? new Person(data.person) : undefined as any;
    this.isDeleted = data.isDeleted;
  }
}
